fiat/fiat-github/src/test/java/com/netflix/spinnaker/fiat/roles/github/GithubTeamsUserRolesProviderTest.java

```diff
@@ -16,66 +16,684 @@
 
 package com.netflix.spinnaker.fiat.roles.github;
 
-import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
-import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
-import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
-import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
-
-import com.github.tomakehurst.wiremock.client.WireMock;
-import com.github.tomakehurst.wiremock.junit5.WireMockExtension;
-import com.netflix.spinnaker.fiat.roles.github.client.GitHubClient;
-import com.netflix.spinnaker.fiat.roles.github.model.Member;
-import com.netflix.spinnaker.kork.retrofit.ErrorHandlingExecutorCallAdapterFactory;
-import com.netflix.spinnaker.kork.retrofit.Retrofit2SyncCall;
-import java.util.List;
-import okhttp3.OkHttpClient;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import com.netflix.spinnaker.fiat.model.resources.Role;
+import com.netflix.spinnaker.fiat.permissions.ExternalUser;
+import java.io.IOException;
+import java.util.*;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.RegisterExtension;
-import retrofit2.Retrofit;
-import retrofit2.converter.jackson.JacksonConverterFactory;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.kohsuke.github.*;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/** Tests for GithubTeamsUserRolesProvider using hub4j/github-api library. */
+@ExtendWith(MockitoExtension.class)
+class GithubTeamsUserRolesProviderTest {
+
+  @Mock private GitHub gitHubClient;
+
+  @Mock private GHOrganization organization;
+
+  @Mock private GHTeam team1;
+
+  @Mock private GHTeam team2;
+
+  @Mock private GHUser user1;
 
-public class GithubTeamsUserRolesProviderTest {
-  @RegisterExtension
-  static WireMockExtension wmGithub =
-      WireMockExtension.newInstance().options(wireMockConfig().dynamicPort()).build();
+  @Mock private GHUser user2;
 
-  GitHubClient gitHubClient;
-  int port;
-  String baseUrl = "http://localhost:PORT/api/v3/";
+  @Mock private PagedIterable<GHUser> orgMembers;
+
+  @Mock private PagedIterable<GHUser> team1Members;
+
+  @Mock private PagedIterable<GHUser> team2Members;
+
+  private GithubTeamsUserRolesProvider provider;
+  private GitHubProperties gitHubProperties;
 
   @BeforeEach
-  void setUp() {
+  void setUp() throws IOException {
```

Comment:
So the ONE disadvantage RIGHT NOW... is this test no longer does REAL API calls against a mock of the github APIs, but does mock of the github client.  Given the library change to use a real GitHub client library, I'm ok with testing that the client is setup correctly.  SO i don't consider this a "negative" but a callout.

Reply:
Sure, added an integration test covering this. It's here: https://github.com/spinnaker/spinnaker/pull/7353/commits/9c909e0299869a5edf7add7338030cc1dcfdee52